lib/mayaUsd/fileio/primUpdaterManager.h

```diff
@@ -78,6 +80,53 @@ class PrimUpdaterManager : public PXR_NS::TfWeakBase
     bool hasPulledPrims() const { return _hasPulledPrims; }
 
 private:
+    struct VariantSelection
```

Comment:
To detect when a change in variant should hide or show a pulled hierarchy, we save the complete list of variant sets and their variant selection along the ancestor chain.  If this variant configuration matches, we show the pulled hierarchy, if it doesn't, we hide the pulled hierarchy.  We cannot rely only on the path, because the path may be identical in two different variants, e.g. all variants of a cup might have a top-level prim called "Cup", so that the path to the selected variant is the same.